### PR TITLE
Add json formatter

### PR DESCRIPTION
**Pull request by harsimranmaan**
Addresses parts of #137

### CODE REVIEW

File: internal/commands/fmt.go [hunk 1]
```diff
@@ -59,11 +67,13 @@ func newFmtCommand(cp configProvider) *cobra.Command {
 	}
 
 	config := fmtCommandConfig{}
-
+	var specifiedTypes []string
 	cmd.Flags().BoolVarP(&config.check, "check-errors", "e", false, "check for unformatted files")
 	cmd.Flags().BoolVarP(&config.write, "write", "w", false, "write result to (source) file instead of stdout")
-	cmd.Flags().BoolVarP(&config.formatJsonnet, "jsonnet", "j", true, "format jsonnet and libsonnet files")
-	cmd.Flags().BoolVar(&config.formatYaml, "yaml", false, "format yaml files")
+	cmd.Flags().StringSliceVar(&specifiedTypes, "types", supportedTypes, "file types that should be formatted")
```
Comment (harsimranmaan) [inline internal/commands/fmt.go:73]: Thinking of dropping the individual flags in favor of this.

**Comment by gotwarlost** [comment]
should we use pretty or would `MarshalIndent` from the `encoding/json` package suffice? Is the main difference key sorting?

**Comment by harsimranmaan** [comment]
Yes, key sorting is the main difference. Gplang's json package does not preserve keys

File: internal/commands/fmt.go (hunk 1 above)
Comment (harsimranmaan) [reply internal/commands/fmt.go:73]: Thoughts on this @gotwarlost ?

**Comment by gotwarlost** [comment]
please rebase from master. I've merged your build changes and fixed the windows build as well

**Pull request merged**